injection/sharedmain/main.go

```diff
@@ -127,6 +127,32 @@ func GetLeaderElectionConfig(ctx context.Context) (*leaderelection.Config, error
 	return leaderelection.NewConfigFromConfigMap(leaderElectionConfigMap)
 }
 
+const (
+	defaultStartInformerDelayTime = 5 * time.Second
+)
+
+type informerStartChanKey struct{}
+type informerStartTimeoutKey struct{}
+
+func WithInformerStart(ctx context.Context, start <-chan struct{}, timeout time.Duration) context.Context {
+	ctx = context.WithValue(ctx, informerStartChanKey{}, start)
+	return context.WithValue(ctx, informerStartTimeoutKey{}, timeout)
+}
+
+func getInformerStartChanTimeout(ctx context.Context) (<-chan struct{}, time.Duration) {
+	timeout := defaultStartInformerDelayTime
+	tuntyped := ctx.Value(informerStartTimeoutKey{})
+	if tuntyped != nil {
+		timeout = tuntyped.(time.Duration)
+	}
+
+	untyped := ctx.Value(informerStartChanKey{})
+	if untyped == nil {
```

Comment:
given return is `interface{}` this might not always work.
https://medium.com/@mangatmodi/go-check-nil-interface-the-right-way-d142776edef1

Reply:
thanks